sdk/identity/azure-identity/src/main/java/com/azure/identity/credential/UserCredential.java

```diff
@@ -0,0 +1,48 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+package com.azure.identity.credential;
+
+import com.azure.core.credentials.AccessToken;
+import com.azure.core.credentials.TokenCredential;
+import com.azure.core.implementation.annotation.Immutable;
+import com.azure.identity.implementation.IdentityClient;
+import com.azure.identity.implementation.IdentityClientBuilder;
+import com.azure.identity.implementation.IdentityClientOptions;
+import reactor.core.publisher.Mono;
+
+import java.util.Objects;
+
+/**
+ * An AAD credential that acquires a token with a username and a password. Users with 2FA/MFA (Multi-factored auth)
+ * turned on will not be able to use this credential. Please use {@link DeviceCodeCredential} or {@link InteractiveBrowserCredential}
+ * instead, or create a service principal if you want to authenticate silently.
+ */
+@Immutable
+public class UserCredential implements TokenCredential {
```

Comment:
>UserCredential [](start = 13, length = 14)

In other languages we have called this UsernamePasswordCredential

Reply:
Renamed